### PR TITLE
Fixing list_roles() and get_role() to work with org-admin

### PR DESCRIPTION
**Pull request by anusuyar**
When login as System admin and  in turn switched to 'orgAdmin', resource_type is 'adminRole'
When loged as 'orgAdmin' directly , resource_type is 'role'

```
C:\sources\github\service_ecosystem\vcd-cli (master)
(vcd-cli) λ vcd login 10.150.200.5 System administrator --no-verify-ssl-certs --password ca$hc0w
InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised.
administrator logged in, org: 'System', vdc: ''

C:\sources\github\service_ecosystem\vcd-cli (master)
(vcd-cli) λ vcd role list
InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised.
isReadOnly    name
------------  ----------------------------
true          System Administrator
true          Multisite System
false         Limited System Administrator

C:\sources\github\service_ecosystem\vcd-cli (master)
(vcd-cli) λ vcd org use STFTestOrg
InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised.
now using org: 'STFTestOrg', vdc: 'vdc-2017-10-18-22-02-32.784'.

C:\sources\github\service_ecosystem\vcd-cli (master)
(vcd-cli) λ vcd role list
InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised.
isReadOnly    name
------------  --------------------------
false         vApp Author
false         Organization Administrator
false         Catalog Author
false         vApp User
false         Console Access Only
true          Defer to Identity Provider

C:\sources\github\service_ecosystem\vcd-cli (master)
(vcd-cli) λ vcd login 10.150.200.5 STFTestOrg pyvcloud --no-verify-ssl-certs --password vmware
InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised.
pyvcloud logged in, org: 'STFTestOrg', vdc: 'vdc-2017-10-18-22-02-32.784'

C:\sources\github\service_ecosystem\vcd-cli (master)
(vcd-cli) λ vcd role list
InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised.
isReadOnly    name
------------  --------------------------
false         vApp Author
false         Organization Administrator
false         Catalog Author
false         vApp User
false         Console Access Only
true          Defer to Identity Provider
```

### CODE REVIEW

File: pyvcloud/vcd/org.py [hunk 1]
```diff
@@ -410,31 +410,39 @@ def create_user(self,
     def list_roles(self):
         if self.resource is None:
             self.resource = self.client.get_resource(self.href)
-        resource_type = 'adminRole'
+        org_filter = None
+        if self.is_admin:
```
Comment (rocknes) [inline pyvcloud/vcd/org.py]: list_role and get_role are almost identical in implementation, we should probably clean them up to minimize code duplication.

Comment (anusuyar) [reply pyvcloud/vcd/org.py]: Done.

File: pyvcloud/vcd/org.py [hunk 2]
```diff
@@ -408,33 +408,33 @@ def create_user(self,
             user)
 
     def list_roles(self):
```
Comment (pacogomez) [inline pyvcloud/vcd/org.py]: brief docstring is missing

File: pyvcloud/vcd/org.py [hunk 3]
```diff
@@ -408,33 +408,33 @@ def create_user(self,
...
-        records = list(q.execute())
-        for r in records:
+        for r in list(roles_query.execute()):
             result.append(to_dict(r,
                                   resource_type=resource_type,
-                                  exclude=['org']))
+                                  exclude=['org', 'orgName', 'href']))
         return result
 
     def get_role(self, role_name):
```
Comment (pacogomez) [inline pyvcloud/vcd/org.py]: brief docstring is missing

File: pyvcloud/vcd/org.py (hunk 2 above)
Comment (anusuyar) [reply pyvcloud/vcd/org.py]: Fixed.

File: pyvcloud/vcd/org.py (hunk 3 above)
Comment (anusuyar) [reply pyvcloud/vcd/org.py]: Fixed.

**Pull request merged**